base/abstractarray.jl

```diff
@@ -2403,7 +2403,14 @@ function hash(A::AbstractArray, h::UInt)
     # Instead hash the tuple of firsts and lasts along each dimension
     h = hash(map(first, axes(A)), h)
     h = hash(map(last, axes(A)), h)
-    isempty(A) && return h
+
+    # For short arrays, it's not worth doing anything complicated 
+    if length(A) < 10000
+        @inbounds for p in pairs(IndexLinear(),A)
```

Comment:
Do we really need to hash the key & value?  Below, too.  This might be an artifact of how we used to do it (which _needed_ to hash the key since we did value-based skipping).  We already hash the axes, and the access pattern will be constant for a given shape.